### DISCUSSION TITLE
Compatability with `io.Closer`

### DISCUSSION THREAD

**Discussion by calebstewart** [klauspost/compress#797] [Q&A]
I noticed today while using the `zstd.Decoder` that it does not properly implement the `io.Closer` interface. The `Close()` method should return an `error`. I'm not sure if this was an oversight or an explicit decision.

Is it possible to add a `error` return value from `Close()` even if it always returns `nil` in order to comply with `io.Closer`?

**Comment by calebstewart** [comment]
I just realized I somehow missed the `IOReadCloser` method, so just wanted to note that I found that.

I'm not going to immediately close this issue, though. I'm curious about the decision to not just change the return type of `Close()` to `error` instead of dealing with a wrapper type. I didn't find any other discussions, although I may have missed them.

Happy to be wrong, but thought I'd ask the question, since just changing the return type seems simpler and doesn't have any negative consequences that I am aware of. :)

**Reply by klauspost** [reply]
The Close doesn't function the same as other `io.Closer` - it makes the Reader unable to be reused. So I didn't want it to accidentally be used as that.

Also changing the type would be a potentially breaking change, so a no-go.

**Reply by calebstewart** [reply]
Understood. Thanks for the explanation! Appreciate you taking the time.
